### PR TITLE
Readme contribution

### PR DESCRIPTION
**Pull request by joanita-51**
### Added Table of Contents and README Sections

This pull request enhances the README by adding a table of contents for easier navigation and adding several sections to provide a more comprehensive project overview. Here's a summary of the changes as suggested in issue #63 

**Table of Contents:** Added a table of contents for quick access to different sections of the README.

**Project Overview**: For a more detailed description of the project.

**Project Benefits:**  To highlight the benefits of this project.

**Images for Other Shapes:** Added images to showcase the various shapes available apart from the one mentioned during installation.

**Contribution:** Contains detailed instructions for potential contributors.

This PR aims to make the README more informative and user-friendly.

### CODE REVIEW

**Comment by joanita-51** [comment]
Hey @TitilayoAdeyemi and @Sudha247, here's my vision for the README on the 'ocaml-joy' project: [README](https://github.com/joanita-51/ocaml-joy/tree/readme-contribution)

Please review and let me know if there's anything I've missed or should remove.

**Comment by TitilayoAdeyemi** [comment]
@joanita-51 I think you did a great job, this is really awesome. Weldone! 
What I think though, is that instead of us to put so many beautiful pictures like you did, we should just put like 6 or 7. A simple one like the circle first, then some more complex ones and the last one should be yet more complex and each of these should have a very short summary under them....It could be just a line describing what the particular shape is/can be used for. I think we should do this so that when prospective users  or contributors take a look at the readme file, they'd be intrigued and want to check out the entire project, kind of like a bait instead of laying out all the pictures. And I think this should be under a subheading with a title like "Shape examples" rather than "Other wonderful shapes" as a continuation of the circle example.

This is what I think. Once again, weldone Joanita! :)

**Comment by joanita-51** [comment]
> @joanita-51 I think you did a great job, this is really awesome. Weldone!
> 
> What I think though, is that instead of us to put so many beautiful pictures like you did, we should just put like 6 or 7. A simple one like the circle first, then some more complex ones and the last one should be yet more complex and each of these should have a very short summary under them....It could be just a line describing what the particular shape is/can be used for. I think we should do this so that when prospective users or contributors take a look at the readme file, they'd be intrigued and want to check out the entire project, kind of like a bait instead of laying out all our pictures. And I think this should be under a subheading with a title like "Shape examples" rather than "Other wonderful shapes" as a continuation of the circle example.
> 
> This is what I think. Once again, weldone Joanita! :)

I agree with that, thanks @TitilayoAdeyemi!

**Comment by nangahamandine** [comment]
> Hey @TitilayoAdeyemi and @Sudha247, here's my vision for the README on the 'ocaml-joy' project: [README](https://github.com/joanita-51/ocaml-joy/tree/readme-contribution)
> 
> Please review and let me know if there's anything I've missed or should remove.

Wow beautifully done @joanita-51!!! You did an amazing job 👍 
Kudos! 💯

**Comment by joanita-51** [comment]
> > Hey @TitilayoAdeyemi and @Sudha247, here's my vision for the README on the 'ocaml-joy' project: [README](https://github.com/joanita-51/ocaml-joy/tree/readme-contribution)
> > Please review and let me know if there's anything I've missed or should remove.
> 
> Wow beautifully done @joanita-51!!! You did an amazing job 👍 Kudos! 💯

Thank you @nangahamandine

File: README.md [hunk 1]
```diff
@@ -1,5 +1,134 @@
...
+1. **Get Creative with Geometry:** This project simplifies geometric creative coding in OCaml, helping you explore the exciting world where programming meets art.
+
+2. **User-Friendly Tools:** We provide an easy-to-use API, so whether you're a newbie or a pro, you can dive into creative coding without confusion.
+
+3. **Basic Shapes at Your Fingertips:** Use simple shapes as your creative building blocks, making it a breeze to start your artistic coding journey.
+
+4. **Mix and Match with Ease:** Combine, transform, and build complex shapes effortlessly.
+
+
+## Installation
```
Comment (Sudha247) [inline README.md:28]: This is quite useful, especially considering this library is aimed to be easy to use for non-OCaml programmers. However, I'd prefer the Installation instructions and the contribution section moved to a separate file, `HACKING.md` or `CONTRIBUTING.md`. The new file can be referenced in the README.

**Review by Sudha247 — CHANGES_REQUESTED** [review]
Thank you @joanita-51! This is a great start. I've suggested some changes below.

File: README.md (hunk 1 above)
Comment (joanita-51) [reply README.md]: Thank you, @Sudha247! I'll start working on the changes right away.

**Comment by joanita-51** [comment]
Thanks, @Sudha247! I have moved the installation and contribution section to CONTRIBUTING.md, referenced in README

**Pull request closed without merging**